tests/unit/test_guess_packager.py

```diff
@@ -0,0 +1,124 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import subprocess
+from collections.abc import Iterable
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+import rpm
+
+from specfile.changelog import guess_packager
+
+
+@pytest.fixture
+def clean_guess_packager(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+) -> Iterable[None]:
+    """
+    Ensure a clean environment
+    """
+    # For $RPM_PACKAGER
+    monkeypatch.delenv("RPM_PACKAGER", False)
+    # Make sure git doesn't read existing config
+    monkeypatch.setenv("HOME", "/dev/null")
+    monkeypatch.delenv("XDG_CONFIG_HOME", False)
+    monkeypatch.chdir(tmp_path)
+    # For %packager
+    old = rpm.expandMacro("%packager")
+    with patch("specfile.changelog._getent_name", return_value=""):
+        try:
+            rpm.delMacro("packager")
+            yield
+        finally:
+            if old != "%packager":
+                rpm.addMacro("packager", old)
+
+
+@pytest.fixture
+def set_packager_env(monkeypatch: pytest.MonkeyPatch) -> str:
+    packager = "Patty Packager <patty@packager.me>"
+    monkeypatch.setenv("RPM_PACKAGER", packager)
+    return packager
+
+
+@pytest.fixture
+def set_packager_git(monkeypatch: pytest.MonkeyPatch, tmp_path: Path) -> Iterable[str]:
+    packager = "Packager, Patty <packager@patty.dev>"
+
+    monkeypatch.chdir(tmp_path)
+    subprocess.run(["git", "init", "."], check=True, capture_output=True)
+    subprocess.run(
+        ["git", "config", "user.name", "Packager, Patty"],
+        check=True,
+        capture_output=True,
+    )
+    subprocess.run(
+        ["git", "config", "user.email", "packager@patty.dev"],
+        check=True,
+        capture_output=True,
+    )
+    return packager
+
+
+@pytest.fixture
+def set_packager_macro() -> Iterable[str]:
+    try:
+        packager = "Patricia Packager"
+        rpm.addMacro("packager", packager)
+        yield packager
+    finally:
+        rpm.delMacro("packager")
+
+
+@pytest.fixture
+def set_packager_passwd() -> Iterable[str]:
+    packager = "Ms. Packager"
+    with patch("specfile.changelog._getent_name", return_value=packager):
```

Comment:
We tend to use [`flexmock`](https://flexmock.readthedocs.io/en/latest/) for mocking. With flexmock, this would become:

```python
flexmock(specfile.changelog).should_receive("_getent_name").and_return(packager)
```

Reply:
I replaced uses of `unittest.mock` with `flexmock`. I *think* I did it correctly 🙂.

Reply:
https://github.com/packit/specfile/pull/220/commits/e4c5bd829e37ea2eed680649e3a0049b3b370a3e